File: services/backend/internal/models/launch-base.go
package models

// for DB
type LaunchBase struct {
	ID         int64 `bson:"id" json:"id,string"`
	ExternalID int64 `bson:"external_id"` // ll2 launch id
}

type LaunchBaseSerializer struct {
	ID   int64                         `json:"id,string"`
	Data LL2LocationSerializerWithPads `json:"data"`
}

type LaunchBaseList struct {
	Count    int                    `json:"count"`
	Launches []LaunchBaseSerializer `json:"launches"`
}

File: apps/admin/src/types/launch-base.ts
import type { LL2LocationSerializerWithPads } from './ll2-location';

export interface LaunchBase {
  id: string;
  external_id: number;
}

export interface LaunchBaseSerializer {
  id: string;
  data: LL2LocationSerializerWithPads;
}

export interface LaunchBaseList {
  count: number;
  launches: LaunchBaseSerializer[];
}
